File: MosPosudit.Model/Requests/Rental/RentalUpdateRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MosPosudit.Model.Requests.Rental
{
    public class RentalUpdateRequest
    {
        [Required(ErrorMessage = "Start date is required")]
        public DateTime StartDate { get; set; }

        [Required(ErrorMessage = "End date is required")]
        public DateTime EndDate { get; set; }

        [Required(ErrorMessage = "Status ID is required")]
        public int StatusId { get; set; }

        public string? Notes { get; set; }
    }
}

File: MosPosudit.UI/shared/lib/dtos/rental/rental_update_request.dart
class RentalUpdateRequestDto {
  final DateTime startDate;
  final DateTime endDate;
  final int statusId;
  final String? notes;

  RentalUpdateRequestDto({
    required this.startDate,
    required this.endDate,
    required this.statusId,
    this.notes,
  });

  Map<String, dynamic> toJson() => {
        'startDate': startDate.toIso8601String(),
        'endDate': endDate.toIso8601String(),
        'statusId': statusId,
        'notes': notes,
      };
}

